src/openstudio_lib/ConstructionObjectVectorController.cpp

```diff
@@ -43,51 +43,27 @@
 #include <openstudio/utilities/idd/IddEnums.hxx>
 
 #include <QMessageBox>
-#include <QTimer>
-#include <QMutex>
 
 namespace openstudio {
 
 // ConstructionObjectVectorController
 
 ConstructionObjectVectorController::ConstructionObjectVectorController(QWidget* parentWidget)
-  : ModelObjectVectorController(), m_reportScheduled(false), m_reportItemsMutex(new QMutex()), m_parentWidget(parentWidget) {}
+  : ModelObjectVectorController(), m_parentWidget(parentWidget) {}
 
-ConstructionObjectVectorController::~ConstructionObjectVectorController() {
-  delete m_reportItemsMutex;
-}
-
-void ConstructionObjectVectorController::reportItemsLater() {
```

Comment:
Only a few `OSVectorController` derived classes were specialized to have the `reportItemsLater` slot which schedules a call to `OSVectorController::reportItems`.  This PR pulls that functionality into the `OSVectorController`.  Now `reportItems` will schedule a call to a `reportItemsImpl` method that will run once so we don't report (and redraw) the same objects multipe times per execute loop.